modules/apps/data-engine/data-engine-js-components-web/test/js/core/utils/sectionAddedHandler.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {default as sectionAdded} from '../../../../src/main/resources/META-INF/resources/js/core/utils/sectionAddedHandler';
+
+const mockPages = [
+	{
+		rows: [
+			{
+				columns: [
+					{
+						fields: [],
+						size: 6,
+					},
+					{
+						fields: [
+							{
+								fieldName: 'FieldsGroup98239067',
+								nestedFields: [
+									{
+										fieldName: 'Date78971332',
+									},
+									{
+										fieldName: 'Numeric21612294',
+									},
+								],
+								rows: [
+									{
+										columns: [
+											{
+												fields: ['Date78971332'],
+												size: 12,
+											},
+										],
+									},
+									{
+										columns: [
+											{
+												fields: ['Numeric21612294'],
+												size: 12,
+											},
+										],
+									},
+								],
+								settingsContext: {
+									pages: [
+										{
+											rows: [
+												{
+													columns: [
+														{
+															fields: [
+																{
+																	fieldName:
+																		'nestedFields',
+																},
+																{
+																	fieldName:
+																		'rows',
+																},
+															],
+															size: 12,
+														},
+													],
+												},
+											],
+										},
+									],
+								},
+								type: 'fieldset',
+							},
+						],
+						size: 6,
+					},
+				],
+			},
+		],
+	},
+];
+
+describe('core/utils/sectionAddedHandler', () => {
+
+	// LPS-143808
+
+	it('add new field by dragging over a fieldset without specifying the location', () => {
+		const {pages} = sectionAdded(
+			{
+				fieldTypes: [
+					{
+						label: 'Fields Group',
+						name: 'fieldset',
+						settingsContext: {
+							pages: [],
+						},
+					},
+				],
+			},
+			{
+				pages: mockPages,
+			},
+			{
+				data: {
+					fieldName: 'FieldsGroup98239067',
+					parentFieldName: 'parentFieldName',
+				},
+				fieldType: {
+					label: 'Text',
+					name: 'text',
+					settingsContext: {
+						pages: [],
+					},
+				},
+				indexes: {
```

Comment:
I think we can delete this `indexes`, since it's irrelevant to your test case. What do you think? Also, do you see any other data in this call that can be deleted for the sake of reducing the number of lines?

Reply:
@rodrigopaulino 
I kept these indexes (that are from the `fieldset`) because it's a copy of what I got when debugging the code and without my fix the test fails as described in the ticket.

Regarding the second question, Initially, I started passing only the relevant data. Additional data was added later because some functions needed it during the test execution. For example, the [createField](https://github.com/liferay/liferay-portal/blob/c78c595de6b96d23310b34c856908ef91a78e5a9/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/utils/sectionAddedHandler.js#L63) requires the `fieldType.settingsContext.pages` [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/fieldSupport.js#L401).

Reply:
It makes sense, @carolmariaabb. Without your fix and without passing `indexes` in your test, would your test fail?

Reply:
Yes, but not with the same error that causes the bug.

It fails because of this error: 
<img width="799" alt="Captura de Tela 2021-12-16 às 11 49 14" src="https://user-images.githubusercontent.com/34186632/146393725-1c61cbfb-07f6-4c3c-a8ee-923c62022206.png">

Reply:
The original error is:
<img width="799" alt="Captura de Tela 2021-12-16 às 11 50 23" src="https://user-images.githubusercontent.com/34186632/146393877-5c36a5b6-4c75-41d3-bf92-2ebda96fbce5.png">

Reply:
I understand. In that case, if you want to guarantee that you're passing the correct hardcoded indexes, then you should mock `fieldSupport`'s `addFieldToPage` and assert the values received as arguments (here's a [doc](https://jestjs.io/docs/mock-functions) that always helps me when I need to do something like that).
Leaving the `indexes` in your test won't guarantee that the same error will occur, because the `fieldSupport` could change while maintaining the same outcome and your test wouldn't have visibility of it. 
Now, if you want to guarantee the same behavior in `fieldSupport`, then you should add a test for `fieldSupport`. 
Does it make sense (please, feel free to disagree. I always like to debate better solutions 😅)?

Reply:
@rodrigopaulino, It makes totally sense!
I just updated the test by creating a spy to confirm the values ​​received as arguments by `addFieldToPage` ([here](https://github.com/liferay-forms/liferay-portal/pull/1262/commits/0dd7674ba5de0f61797ca8603805a7d121e7f554#diff-817dd8b054fff445c3c205980137199f3fc69e480f9c1b7a2705501af2cfd743R132-R142)).

What do you think?

Reply:
It looks great, @carolmariaabb! 🎉 
I think we can resolve this topic.